plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+                ap_selected_field = self.payload.get("config")[0].get("ap_selected_field")
+                if ap_selected_field is None or ap_selected_field == "" or ap_selected_field == "all":
+                    self.payload["device_config"] = ap_response
+                else:
+                    self.payload["device_config"]=self.data_frame(ap_selected_field,[ap_response])
+
+                self.log("AP configuration {0} exists in Cisco Catalyst Center"\
+                         .format(str(self.payload["device_config"][0])), "INFO")
+                accesspoint_exists = True
+
+        return (accesspoint_exists, current_configuration)
+
+    def compare_ap_cofig_with_inputdata(self, apconfig):
+        """
+        This function used to compare with the input ap detail with the current ap configuration
+        information are not same, those data will be updated in the AP input information.
+        Parameters:
+          - apconfig: This is response of the get_have
+        Returns:
+            This will be the return the final data for update AP detail.
+            "final_input": [
+                {
+                    "adminStatus": true,
+                    "apList": [
+                        {
+                            "macAddress": "34:5d:a8:0e:20:b4"
+                        }
+                    ],
+                    "configureLedBrightnessLevel": true,
+                    "ledBrightnessLevel": 4,
+                    "macAddress": "34:5d:a8:0e:20:b4"
+                } ]
+        Example:
+            functions = Accesspoint(module)
+            final_input_data = functions.compare_ap_cofig_with_inputdata(all_apconfig)
+        """
+
+        newdict = {}
+        if self.want and apconfig:
+            if self.want.get("mac_address") == apconfig["mac_address"] or \
+                    self.want.get("hostname") == apconfig["ap_name"]:
+                allkey = list(self.want.keys())
+                for value in ("mac_address","hostname","management_ip_address", "family"):
+                    if value in allkey: allkey.remove(value)
+                for each_key in allkey:
+                    if each_key == "ap_name_new":
+                        if self.want["ap_name_new"] != apconfig.get("ap_name"):
+                            newdict["apNameNew"] = self.want["ap_name_new"]
+                    elif each_key == "ap_name":
+                        newdict[self.keymap[each_key]] = self.want[each_key]
+                    else:
+                        if self.want[each_key] != apconfig[each_key]:
+                            newdict[self.keymap[each_key]] = self.want[each_key]
+                if newdict.get("apName") is not None and newdict.get("apNameNew") is None:
+                    del newdict["apName"]
+                if newdict:
+                    newdict["macAddress"] = apconfig["eth_mac"]
+            if newdict:
+                return newdict
+            else:
```

Comment:
this else is not needed if you return from 'if' block

Reply:
Here else need to track same data is DNAC so need to track log as well as need to return none.

Reply:
As Abhishek pointed out, we do not need an "else" statement. If "update_config" is not present, it is obvious that we will return None with log message..

```
            if update_config:
                self.log("....")
                return update_config
            
            self.log('Playbook AP Configuration remain same in Current AP configration',
                      "INFO")
            return None
```